app/models/miq_queue_worker_base/runner.rb

```diff
@@ -106,7 +106,9 @@ def deliver_queue_message(msg)
       if status == MiqQueue::STATUS_TIMEOUT
         begin
           _log.info("#{log_prefix} Reconnecting to DB after timeout error during queue deliver")
-          ActiveRecord::Base.connection.reconnect!
+
+          # Remove the connection and establish a new one since reconnect! doesn't always play nice with SSL postgresql connections
+          ActiveRecord::Base.establish_connection(ActiveRecord::Base.remove_connection("primary"))
```

Comment:
Do we need to explicitly specify the primary connection name or will this work without a parameter?

Reply:
yeah, it needs to [disconnect the pool and drop the key for the specification name](https://github.com/rails/rails/blob/2e824afc06011d5f75a7c964be007589771c89a5/activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb#L899) from the `owner_to_pool` hash:

Reply:
But I don't think that's the `remove_connection` we call.

```ruby
irb(main):001:0> ActiveRecord::Base.method(:remove_connection).source_location
=> ["/home/ncarboni/.gem/ruby/2.4.3/gems/activerecord-5.0.7/lib/active_record/connection_handling.rb", 136]
```
Or is `@connection_specification_name` not `"primary`" [here](https://github.com/rails/rails/blob/2e824afc06011d5f75a7c964be007589771c89a5/activerecord/lib/active_record/connection_handling.rb#L137)

Reply:
Yeah, that should work too.  Good point.  

Note, I chose the hardcoded "primary" because these places were calling explicitly `reconnect!` on `AR::Base` and not a model/subclass.  I guess calling `connection_specification_name` on `AR::Base` is nicer than the hardcoded "primary".  I'd rather not rely on `connection_handler.retrieve_connection_pool(name)` returning "primary" when the name is nil though.

What do you think?

```
spec_name = ActiveRecord::Base.connection_specification_name
ActiveRecord::Base.establish_connection(ActiveRecord::Base.remove_connection(spec_name))
```

Reply:
Yeah, that looks good.

It does seem a bit strange that they choose to go straight to the instance variable in `remove_connection` rather than using the logic in the `connection_specification_name` method. I'm sure there are __Reasons__ for the way it's done, but your version looks more like the behavior we want here.

Reply:
The version you wrote up in the comment that is.... Just realized that was a bit ambiguous as both versions are "yours"

Reply:
😆 

Note, I was going to one line to avoid adding more lines but 3 AR::Base on one line is one too many.